### PR TITLE
ledger: Update onlineAccounts lookup methods for slightly easier reading

### PR DESCRIPTION
**Pull request by cce**
## Summary

Was reviewing onlineAccounts.lookupOnlineAccountData and onlineTop with @algoidan and @algorandskiy and made some changes for comments and clarity.

## Test Plan

Existing tests should pass.

### CODE REVIEW

File: ledger/acctonline.go [hunk 1]
```diff
@@ -656,23 +656,20 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
 			// no such online account, return empty
 			return ledgercore.OnlineAccountData{}, err
 		}
-		// lookupOnlineHistory does not fetch the account db round because of the following observation:
-		// 1. ao.onlineAccountsCache update happens with ao.accountsMu taken below and in postCommit
-		// 2. If we started reading the history (lookupOnlineHistory)
-		//   1. before commitRound or while it is running => OK, read what is in DB and then add new entries in postCommit
```
Comment (algorandskiy) [inline ledger/acctonline.go:662]: I'd like to preserve these cases consideration b/c it explains logic when we need to retry and etc

File: ledger/acctonline.go [hunk 2]
```diff
@@ -656,23 +656,20 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
...
+		// this online account will not hit the on-disk DB.
 		persistedDataHistory, validThrough, err := ao.accountsq.lookupOnlineHistory(addr)
 		if err != nil || len(persistedDataHistory) == 0 {
 			return ledgercore.OnlineAccountData{}, err
 		}
-		// 3. If we fishied reading the history (lookupOnlineHistory)
+		// onlineAccountsCache update happens with ao.accountsMu taken below, and in postCommit
+		// After we finished reading the history (lookupOnlineHistory), either
 		//   1. db has not advanced (validThrough == currentDbRound) => OK
 		//   2. after commitRound but before postCommit (currentDeltaLen == len(ao.deltas)) => OK, the cache gets populated and postCommit updates the new entry
```
Comment (algoidan) [inline ledger/acctonline.go:669]: should we include `currentDbRound >= ao.cachedDBRoundOnline` condition as well? it is part of the post-commit check

**Review by algoidan — APPROVED** [review]

File: ledger/acctonline.go (hunk 1 above)
Comment (cce) [reply ledger/acctonline.go:662]: OK, updated

File: ledger/acctonline.go (hunk 2 above)
Comment (cce) [reply ledger/acctonline.go]: updated

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**